docs/providers

```diff
@@ -99,11 +99,11 @@ htonll() / ntohll()
 sizeof_field()
 MIN / MAX - simple (i.e. potentially unsafe) min/max macros
 roundup_power_of_two()
-fastlock_t - lock object (spinlock or mutex based on availability)
+ofi_spin_t - lock object (spinlock or mutex based on availability)
```

Comment:
this line needs to be split.

Reply:
This whole file needs to be updated at this point, but the spin lock comment still holds for non-Linux platforms.  I'll look at updating the file with more generic descriptions of what's available in a separate PR.